src/main/java/uk/gov/justice/hmpps/nomis/datacompliance/repository/jpa/repository/DuplicateOffenderRepository.java

```diff
@@ -69,4 +69,150 @@
             ")",
             nativeQuery = true)
     List<DuplicateOffender> getOffendersWithMatchingLidsNumbers(String offenderNo);
+
+    /**
+     * The following query identifies potential duplicate offender records
+     * based on matching personal details. Key components of this query are
+     * as follows:
+     *
+     * - The Jaro Winkler algorithm is used to fuzzy match details to avoid
+     *   misspellings or typos affecting the result
+     *
+     * - The match is points based, with certain fields having a higher
+     *   weighting
+     *
+     * - All offender aliases are taken into account and for each field the
+     *   values with maximum similarity are used towards the points calculation
+     *
+     * - The weightings have been chosen based on empirical evidence
+     */
+    @Query(value =
+            "WITH OFFENDER_MATCH_DETAILS AS (" +
+            "    SELECT *" +
+            "    FROM MV_OFFENDER_MATCH_DETAILS" +
+            "    WHERE OFFENDER_ID_DISPLAY = :offenderNo" +
+            "), " +
+            "SIMILARITY AS (" +
+            "    SELECT" +
+            "        O.OFFENDER_ID_DISPLAY AS OFFENDER_ID_DISPLAY," +
+            "        (CASE WHEN OMD.AVG_HEIGHT = 0 THEN 0 ELSE ((OMD.AVG_HEIGHT - ABS(O.AVG_HEIGHT - OMD.AVG_HEIGHT)) * 100 / OMD.AVG_HEIGHT) END) AS HEIGHT_SIMILARITY," +
+            "        UTL_MATCH.JARO_WINKLER_SIMILARITY (O.LAST_NAME_KEY, OMD.LAST_NAME_KEY) AS LAST_NAME_SIMILARITY," +
+            "        UTL_MATCH.JARO_WINKLER_SIMILARITY (O.FIRST_NAME_KEY, OMD.FIRST_NAME_KEY) AS FIRST_NAME_SIMILARITY," +
+            "        (CASE WHEN ABS(MONTHS_BETWEEN(O.BIRTH_DATE, OMD.BIRTH_DATE)) >= 120 THEN 0" +
+            "         ELSE UTL_MATCH.JARO_WINKLER_SIMILARITY (O.BIRTH_DATE, OMD.BIRTH_DATE) END" +
```

Comment:
Didn't know you could do a Wrinkler similarity with dates, I assume it convert them to Strings?

Reply:
Yes it does just convert to strings to do the match.  I think it is more heavily penalising month differences because of that so I will look to fix that.